.github/workflows/build-docs.yml

```diff
@@ -1,21 +1,39 @@
 name: Build documentation
-on: workflow_dispatch
+
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: write
 
 jobs:
   build-docs:
     runs-on: ubuntu-20.04
     container:
       image: qgis/qgis:release-3_16
     steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+
       - name: Checkout code
         uses: actions/checkout@v2
+        with:
+          key: ${{github.ref}}
+          path: .cache
 
       - name: Fix Python command
         run: apt-get install python-is-python3
 
       - name: Install plugin dependencies
         run: pip install -r requirements-dev.txt
 
+      - name: Install mkdocs-material
```

Comment:
We don't need this step as it is already available in the requirements-dev file

Reply:
Done